garak/resources/gcg/generate_gcg.py

```diff
@@ -64,7 +65,7 @@
 gcg_parser.add_argument(
     "--outfile",
     type=str,
-    default=gcg_resource_data / "gcg_prompts.txt",
+    default=resource_data / "gcg_prompts.txt",
```

Comment:
Since this is output, should this go to the cache location by default?

```suggestion
    default=gcg_resource_data / "gcg_prompts.txt",
```

Thoughts?

Reply:
Thinking more on this, it may be reasonable to create utility functions for read and write of resource files, and enforce a coding standard of access where a `read` file handle to a `resource` is returned based on two levels of access, first searching for the file in `XDG_DATA_HOME/garak/PATH_TO_RESOURCE_FILE` and falling back to return `BASEDIR/PATH_TO_RESOURCE_FILE` when not found. As a standard coding practice for the project all `resources` shipped with the project must be accessed through this utility.

Extending from this the standard for `write` to resource files would alway place the file in `XDG_DATA_HOME/garak` to ensure the tool does not write to files in the installation path, but rather in the a user owned location. An additional benefit of this would be that the user could then override any `resource` file by placing their own version in `XDG_DATA_HOME/garak/PATH_TO_RESOURCE_FILE`.  An immediate benefit of this would be see in the `suffix.GCGCached` probe by allowing the cached set of prompts to be overwritten using this pattern. This could be further extended to centralize `common` as loadable files not embedded as strings in the codebase ensuring less churn unrelated to logic in the `python` files.

Reply:
Yes out of scope here, I have ensured `output` files are consistently in `cache_dir` for now.